plot_features.py

```diff
@@ -132,40 +142,45 @@ def plot_features_file(featurefile, mapInfo): #{{{
     # use colorbrewer qualitative 7 data class colors, "7-class Accent": http://colorbrewer2.org/
     colors = ['#7fc97f' ,'#beaed4', '#fdc086', '#ffff99','#386cb0','#f0027f','#bf5b17']
     markers = ['o', 's', 'v', '^', '>', '<', '*', 'p', 'D', 'h']
+    linestyle = ['-', '--', '-.', ':']
 
     feature_num = 0
 
     for feature in featuredat['features']:
         polytype = feature['geometry']['type']
         coords = feature['geometry']['coordinates']
-        feature = feature['properties']['name']
-        print '  feature: %s'%feature
+        shape = shapely.geometry.shape(feature['geometry'])
+        featurename = feature['properties']['name']
+        print '  feature: %s'%featurename
 
         color_num = feature_num % len(colors)
         marker_num = feature_num % len(markers)
+        linestyle_num = feature_num % len(linestyle)
 
         try:
             if polytype == 'MultiPolygon':
+                for apoly in shape:
+                    plot_feature(mapInfo, 'polygon', apoly, color=colors[color_num])
+            elif polytype == 'Polygon':
+                plot_feature(mapInfo, 'polygon', shape, color=colors[color_num])
+            elif polytype == 'MultiLineString':
                 for poly in coords:
-                    for shape in poly:
-                        plot_poly(mapInfo,shape,colors[color_num])
-            elif polytype == 'Polygon' or polytype == 'MultiLineString':
-                for poly in coords:
-                    plot_poly(mapInfo,poly,colors[color_num])
+                    plot_feature(mapInfo, 'line', poly, color=colors[color_num], linestyle=linestyle[linestyle_num])
             elif polytype == 'LineString':
-                plot_poly(mapInfo,coords,colors[color_num],filled=False)
+                plot_feature(mapInfo, 'line', coords, color=colors[color_num], linestyle=linestyle[linestyle_num])
             elif polytype == 'Point':
-                plot_point(mapInfo,coords,markers[marker_num],colors[color_num])
+                plot_feature(mapInfo, 'point', coords, color=colors[color_num], marker=markers[marker_num])
             else:
                 assert False, 'Geometry %s not known.'%(polytype)
-        except:
-            print 'Error plotting %s'%(feature)
+        except Exception as inst:
+            print 'Error plotting %s with error %s'%(featurename, inst)
 
         feature_num = feature_num + 1
 
     for mapType in mapInfo:
         (ax, projection, plotFileName, fig) = mapInfo[mapType]
         print 'saving ' + plotFileName
+        ax.set_global()
```

Comment:
@pwolfram, am I right in thinking that this line has made it so the plots don't "zoom" in on the region being plotted but rather plot the entire globe every time?

I'm okay with that change but I want to make sure others are as well.  i could imagine cases where a given plot is quite regional and plotting the entire globe is overkill (e.g. when I've been focused on the Mediterranean) so maybe we want a command-line argument that lets the user force a plot to be "regional" if they want that?